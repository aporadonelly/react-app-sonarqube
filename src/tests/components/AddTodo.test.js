import React from "react";
import "jest-enzyme";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AddTodo from "../../components/AddTodo";

configure({ adapter: new Adapter() });

describe("TodoItem", () => {
  it("should display todoitem with no error", () => {
    const wrapper = shallow(<AddTodo />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should handle submit handler", () => {
    const handleAddTodo = jest.fn();
    const wrapper = shallow(<AddTodo handleAddTodo={handleAddTodo} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.find("Form").simulate("submit");
    expect(handleAddTodo).toHaveBeenCalled();
  });
});
