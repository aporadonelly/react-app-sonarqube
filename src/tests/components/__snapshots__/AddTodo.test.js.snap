// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem Should handle submit handler 1`] = `
<Form
  inline={false}
  onSubmit={[MockFunction]}
>
  <FormGroup
    controlId="item"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      What you want to do?
    </FormLabel>
    <FormControl
      name="todo"
      placeholder="Enter todo"
      type="text"
    />
  </FormGroup>
  <Button
    active={false}
    disabled={false}
    type="submit"
    variant="primary"
  >
    Add Todo
  </Button>
</Form>
`;

exports[`TodoItem should display todoitem with no error 1`] = `
<Form
  inline={false}
>
  <FormGroup
    controlId="item"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      What you want to do?
    </FormLabel>
    <FormControl
      name="todo"
      placeholder="Enter todo"
      type="text"
    />
  </FormGroup>
  <Button
    active={false}
    disabled={false}
    type="submit"
    variant="primary"
  >
    Add Todo
  </Button>
</Form>
`;
